presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -111,17 +113,25 @@ public Expression rewriteSymbolReference(SymbolReference node, Void context, Exp
         }, value);
     }
 
-    public OrderingScheme map(OrderingScheme orderingScheme)
+    public RowExpression map(RowExpression value)
     {
-        return new OrderingScheme(orderingScheme.getOrderBy().stream().map(this::map).collect(toImmutableList()),
-                orderingScheme.getOrderings().entrySet().stream().collect(toMap(entry -> map(entry.getKey()), entry -> entry.getValue())));
+        if (isExpression(value)) {
```

Comment:
let's move this part of logic to the caller. The caller should decide which method to use.

Reply:
SymbolMapper will be used to in multiple optimization rules. As we are pushing the clean up up, we will need this function to handle both Expression/RowExpression anyway. I would prefer we just remove isExpression later once everything is cleaned up.